core/services/feeds/service.go

```diff
@@ -285,14 +285,22 @@ func (s *service) SyncNodeInfo(ctx context.Context, id int64) error {
 
 // UpdateManager updates the feed manager details, takes down the
 // connection and reestablishes a new connection with the updated public key.
-func (s *service) UpdateManager(ctx context.Context, mgr FeedsManager) error {
+func (s *service) UpdateManager(ctx context.Context, mgr PartialFeedsManager) error {
 	err := s.orm.UpdateManager(ctx, mgr)
 	if err != nil {
 		return errors.Wrap(err, "could not update manager")
 	}
 
-	if err := s.restartConnection(ctx, mgr); err != nil {
-		s.lggr.Errorf("could not restart FMS connection: %v", err)
+	if mgr.IsEnabled == nil || *mgr.IsEnabled {
```

Comment:
what happens if the current enabled state is false, and then user call the Update Manager api without providing isEnabled value, will it try to restart the connection?

Reply:
Yes. This is the current behavior for any updateFeedsManager call currently as well.

Reply:
i think there is a bug here, if my existing state is disabled, and i call updateManger by setting isEnabled to nil, this logic will starts to reconnect to manager which we dont want right?

Reply:
The nil value is used to maintain backward compatibility, as changing this behavior would introduce a breaking change. When using the UI, users will always see the current state, so they shouldn't pass nil unintentionally. After the introduction of multiple Job Distributors (JDs), users will be aware of the new field and are unlikely to pass null in this context. What do you think?